src/logics/Logic.cc

```diff
@@ -1600,3 +1607,108 @@ char* Logic::printTerm        (PTRef tr, bool l, bool s) const { return printTer
 void Logic::termSort(vec<PTRef>& v) const { sort(v, LessThan_PTRef()); }
 
 void  Logic::purify     (PTRef r, PTRef& p, lbool& sgn) const {p = r; sgn = l_True; while (isNot(p)) { sgn = sgn^1; p = getPterm(p)[0]; }}
+
+bool Logic::typeCheck(SymRef sym, vec<PTRef> const & args, std::string & why) const {
+
+    auto genSortMismatchString = [&](SymRef sym, vec<PTRef> const & args) {
+        std::string symStr = getSymName(sym);
+        Symbol const & symbol = sym_store[sym];
+        if (symbol.chainable() or symbol.pairwise()) {
+            for (auto _ : args) {
+                (void)_;
+                symStr += " " + std::string(getSortName(symbol[0]));
+            }
```

Comment:
The raw loop seems better fitting in this case, since you don't want to do anything about the elements, just repeat some action a given number of times.

```
for (int i = 0; i < args.size(); ++i) {
    symStr += " " + std::string(getSortName(symbol[0]));
}
```

Especially for the later loops, where constructing the span seems unnecessary when a simple counter will do.

Reply:
Well, it's kind of for each element.  But anyway, I'll change them to counters.